src/main/java/com/woowacamp/soolsool/core/liquor/service/LiquorService.java

```diff
@@ -83,56 +83,60 @@ public LiquorDetailResponse liquorDetail(final Long liquorId) {
     }
 
     @Transactional
-    public List<LiquorElementResponse> liquorList(
+    public PageLiquorResponse liquorList(
         final LiquorBrewType brewType,
         final LiquorRegionType regionType,
         final LiquorStatusType statusType,
         final String brand,
-        final Pageable pageable
+        final Pageable pageable,
+        final Long cursorId
     ) {
-        final Specification<Liquor> conditions =
-            searchWith(brewType, regionType, statusType, brand);
+        final LiquorSearchCondition liquorSearchCondition = new LiquorSearchCondition
+            (
+                findLiquorRegionByType(regionType),
+                findLiquorBrewByType(brewType),
+                findLiquorStatusByType(statusType),
+                brand
+            );
 
-        final Page<Liquor> liquors = liquorRepository.findAll(conditions, pageable);
+        final List<LiquorElementResponse> liquors = liquorQueryDslRepository
+            .getList(liquorSearchCondition, pageable, cursorId);
 
         liquorCtrRepository.findAllByLiquorIdInWithPessimisticWriteLock(extractLiquorIds(liquors))
             .forEach(LiquorCtr::increaseImpressionOne);
 
-        return LiquorElementResponse.from(liquors);
+        return getPageLiquorResponse(pageable, liquors);
     }
 
-    private Specification<Liquor> searchWith(
-        final LiquorBrewType brewType,
-        final LiquorRegionType regionType,
-        final LiquorStatusType statusType,
-        final String brand
-    ) {
-        return ((root, query, criteriaBuilder) -> {
-            final List<Predicate> predicates = new ArrayList<>();
-
-            findLiquorBrewByType(brewType).ifPresent(liquorBrew -> predicates
-                .add(criteriaBuilder.equal(root.get("brew"), liquorBrew)));
+    @Transactional
+    public PageLiquorResponse getFirstPage(final Pageable pageable) {
+        final List<LiquorElementResponse> liquors = liquorQueryDslRepository
+            .getCachedList(pageable);
 
-            findLiquorRegionByType(regionType).ifPresent(liquorRegion -> predicates
-                .add(criteriaBuilder.equal(root.get("region"), liquorRegion)));
+        liquorCtrRepository.findAllByLiquorIdInWithPessimisticWriteLock(extractLiquorIds(liquors))
+            .forEach(LiquorCtr::increaseImpressionOne);
 
-            findLiquorStatusByType(statusType).ifPresent(liquorStatus -> predicates
-                .add(criteriaBuilder.equal(root.get("status"), liquorStatus)));
+        return getPageLiquorResponse(pageable, liquors);
+    }
 
-            if (StringUtils.hasText(brand)) {
-                predicates.add(criteriaBuilder.equal(root.get("brand"), new LiquorBrand(brand)));
-            }
+    private PageLiquorResponse getPageLiquorResponse(
+        final Pageable pageable,
+        final List<LiquorElementResponse> liquors
+    ) {
+        if (liquors.size() < pageable.getPageSize()) {
+            return PageLiquorResponse.of(false, null, liquors);
```

Comment:
```suggestion
            return PageLiquorResponse.of(false, liquors);
```

null을 없애고 of를 오버로드 하는게 좋을 것 같아요.

Reply:
햇숨니다